app/controllers/authentications_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class AuthenticationsController < ApplicationController
+  before_action :authorize_request, except: :create
+
+  # Post /auth/login
+  def create
+    @user = User.find_by(email: params[:email])
+    puts @user
+    puts params[:password]
+    if @user&.authenticate(params[:password])
+      token = JsonWebToken.encode(user_id: @user.id)
+      time = Time.zone.now + 24.hours.to_i
+      render json: { token:, exp: time.strftime('%m-%d-%Y %H:%M'),
+                     email: @user.email }, status: :ok
```

Comment:
I'd suggest to have `JsonWebToken.encode` to return a hash with `token` and `exp`, this way you don't have to calculate the expiration time from the controller.

Then you would have something like
```suggestion
      jwt = JsonWebToken.encode(user_id: @user.id)
      render json: { email: @user.email, **jwt }, status: :ok
```

Reply:
This one, was resolved today. I'll mark this comment as such!